docs/canonicalk8s/_parts/common_hardening.md

```diff
@@ -1,10 +1,83 @@
 ### Control plane nodes
 
+#### Control Plane Security
+
+Kubernetes documentation on [controlling access][controlling_access] to the API informs on topics
+such as **Transport Security**, **AAA (Authentication,Authorization,Auditing)**,
+and **Admission Control**. Much of this discusses how to limit access to
+authorized users and disable access to unauthorized users.
+
+#### Role Based Access Control
+
+Kubernetes documentation on [rbac][access_authn_authz] informs how to
+enforce RBAC policies to limit access to cluster resources, and define roles
+and permissions aligned with the principle of least privilege to prevent
+unauthorized access to critical operations.
+
+#### Secrets Encryption at Rest
+Kubernetes documentation on [encrypting secrets][encryption_at_rest] informs how
+to ensure that the key-value store encrypts secrets rather than leaving it as
+base64 encoded values.
+
+this guide indicates one should set the `--encryption-provider-config` file
+as an argument to the kubernetes apiserver.
+
+Create the `EncryptionConfiguration` file under `/var/snap/k8s/common/etc/encryption/`
+and update the kube-apiserver arguments.
+
+```
+sudo sh -c '
+mkdir -p /var/snap/k8s/common/etc/encryption/
+cat >/var/snap/k8s/common/etc/encryption/enc.yaml << EOL
+kind: "EncryptionConfig"
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+- resources: ["secrets"]
+  providers:
+  - aesgcm:
+      keys:
+      - name: key1
+        secret: ${BASE 64 ENCODED SECRET}
+  - identity: {}
+EOL
+chmod 600 /var/snap/k8s/common/etc/encryption/enc.yaml
+cat >>/var/snap/k8s/common/args/kube-apiserver <<EOL
+--encyrption-provider-config=/var/snap/k8s/common/etc/enc.yaml
+EOL'
+```
+
+This ensures that cluster `secrets` resources are encrypted in the key-value
+store. Securing the contents of this key file is left as a separate exercise.
+
+
+#### Configuring Authorization Modes
+Enforce RBAC (Role-Based Access Control) policies to limit access to cluster
+resources. Define roles and permissions aligned with the principle of least
+privilege to prevent unauthorized access to critical operations.
+
+Confirm the value of the apiserver [`authorization-mode`][authorization_mode]
+value
+* includes `RBAC`
+* doesn't include `AlwaysAllow`
+
+```
+sudo grep authorization-mode /var/snap/k8s/common/args/kube-apiserver | \
+    grep -q "RBAC" && echo "okay" || echo "missing"
+sudo grep authorization-mode /var/snap/k8s/common/args/kube-apiserver | \
+    grep -q "AlwaysAllow" && echo "Remove AlwaysAllow" || echo "okay"
+```
+
+By default, the value is `Node,RBAC`
+* `Node`: 
+  A special-purpose authorization mode that grants permissions
+  to kubelets based on the pods they are scheduled to run.
+
+
```

Comment:
Suggestion: Remove the RBAC section above and simply include here:
To apply RBAC to other cluster resources, see the upstream Kubernetes [RBAC guide][access_authn_authz].